code/modules/shuttle/on_move.dm

```diff
@@ -1,40 +1,273 @@
+	//rotate our direction
+	setDir(angle2dir(rotation+dir2angle(dir)))
+
+	//resmooth if need be.
+	if(smooth)
+		queue_smooth(src)
+
+	//rotate the pixel offsets too.
+	if (pixel_x || pixel_y)
+		if (rotation < 0)
+			rotation += 360
+		for (var/turntimes=rotation/90;turntimes>0;turntimes--)
+			var/oldPX = pixel_x
+			var/oldPY = pixel_y
+			pixel_x = oldPY
+			pixel_y = (oldPX*(-1))
+
+/************************************Turf move procs************************************/
+
+/turf/open/afterShuttleMove(turf/oldT) //Recalculates SSair stuff for turfs on both sides
 	. = ..()
+	SSair.remove_from_active(src)
+	SSair.remove_from_active(oldT)
 
+	src.CalculateAdjacentTurfs()
+	oldT.CalculateAdjacentTurfs()
 
-/atom/movable/light/onShuttleMove()
-	return 0
+	SSair.add_to_active(src, TRUE)
+	SSair.add_to_active(oldT, TRUE)
 
-/obj/machinery/door/airlock/onShuttleMove()
+/************************************Machinery move procs************************************/
+
+/obj/machinery/door/airlock/beforeShuttleMove()
+	. = ..()
 	shuttledocked = 0
 	for(var/obj/machinery/door/airlock/A in range(1, src))
 		A.shuttledocked = 0
 		A.air_tight = TRUE
 		INVOKE_ASYNC(A, /obj/machinery/door/.proc/close)
+
+/obj/machinery/door/airlock/afterShuttleMove()
 	. = ..()
 	shuttledocked =  1
 	for(var/obj/machinery/door/airlock/A in range(1, src))
 		A.shuttledocked = 1
+
+/obj/machinery/camera/beforeShuttleMove()
+	. = ..()
+	GLOB.cameranet.removeCamera(src)
+	GLOB.cameranet.updateChunk()
+	return TRUE
+
+/obj/machinery/camera/afterShuttleMove()
+	. = ..()
+	if(can_use())
+		GLOB.cameranet.addCamera(src)
+	var/datum/camerachunk/chunk = GLOB.cameranet.getCameraChunk(x, y, z)
+	chunk.hasChanged(TRUE)
+
+/obj/machinery/telecomms/onShuttleMove(turf/T1)
+	. = ..()
+	if(. && T1) // Update listening Z, just in case you have telecomm relay on a shuttle
+		listening_level = T1.z
+
+/obj/machinery/mech_bay_recharge_port/afterShuttleMove()
+	. = ..()
+	recharging_turf = get_step(loc, dir)
+
+/obj/machinery/atmospherics/afterShuttleMove()
+	. = ..()
+	if(pipe_vision_img)
+		pipe_vision_img.loc = loc
+
+/obj/machinery/computer/auxillary_base/onShuttleMove(turf/T1)
+	. = ..()
+	if(z == ZLEVEL_MINING) //Avoids double logging and landing on other Z-levels due to badminnery
+		SSblackbox.add_details("colonies_dropped", "[x]|[y]|[z]") //Number of times a base has been dropped!
+
+/obj/machinery/gravity_generator/main/beforeShuttleMove()
+	. = ..()
+	on = FALSE
+	update_list()
+
+/obj/machinery/gravity_generator/main/afterShuttleMove()
+	. = ..()
+	if(charge_count != 0 && charging_state != POWER_UP)
+		on = TRUE
+	update_list()
+
+/obj/machinery/thruster/beforeShuttleMove()
+	. = ..()
+	. = TRUE
+
+//Properly updates pipes on shuttle movement
+/obj/machinery/atmospherics/shuttleRotate(rotation)
+	var/list/real_node_connect = getNodeConnects()
+	for(DEVICE_TYPE_LOOP)
+		real_node_connect[I] = angle2dir(rotation+dir2angle(real_node_connect[I]))
+
+	. = ..()
```

Comment:
It needs to do things before and after the base shuttleRotate proc is my guess. I didn't write this particular bit, it's a port over from ftl13